server/pairing/TODO.MD

```diff
@@ -0,0 +1,128 @@
+# Local Pairing To-dos
+
+- [Network Address / Peer Discovery](#network-address--peer-discovery)
+- [Network Debug Diagnostics](#network-debug-diagnostics)
+
+## Network Address / Peer Discovery
+
+Revamp the preferred network discovery process
+
+Note this functionality is pre-secret share so doesn't ...
+
+- **Pre-shared secret: scan the QR code first, containing the secret and then attempt to discover the other device(s)**
+  - ⚠️ **Problem:** Only works if UDP discovery works. [See possible solution for UDP discovery failure mitigation](#udp-discovery-failure-mitigation)
+  - Start pairing server on relevant device, listen to `0.0.0.0`.
+    - get selected port number
+    - get TLS pub key
+    - get secret
+  - Scan QR code which including:
+    - selected port number
+    - TLS pub key
+    - secret
+  - UDP discovery happens 🧙🪄✨
+  - devices acquire ip addresses of peer(s), and themselves.
+  - client device attempts connection as normal
+- **Post-shared secret, which is confirmed later with the QR code**
+  - UDP discovery happens 🧙🪄✨
+  - devices acquire ip addresses of peer(s), and themselves.
+    - can also optionally give feedback to the user with the pairing options.
+  - server device starts server listener, listen to known network ip address
+    - get selected port number
+    - get TLS pub key
+    - get secret
+  - Scan QR code which including:
+    - selected port number
+    - TLS pub key
+    - secret
+  - client device attempts connection as normal
+
+### Self IP address acquisition
+
+1. Ask network interface for its preferred outbound IP address
```

Comment:
This doesn't sound quite correct to me.
What we currently do is dial `255.255.255.255:8080` and check the local IP address for the connection.

https://github.com/status-im/status-go/blob/dbb4369de1381e267019cb27c25775ec51c14991/server/ips.go#L12-L22

So we rely on OS to select best network interface for reaching `255.255.255.255`. I think this way we target the *default* network interface of OS, which would be set in OS's routing table.

So to say it correct it would be smth like `Get preferred IP address of OS default network interface`.